BillTrack.Persistence/Configuration/EmployeeConfiguration.cs

```diff
@@ -0,0 +1,58 @@
+using BillTrack.Domain.Entities;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.EntityFrameworkCore.Metadata.Builders;
+
+namespace BillTrack.Persistence.Configuration;
+
+public class EmployeeConfiguration : IEntityTypeConfiguration<Employee>
+{
+    public void Configure(EntityTypeBuilder<Employee> builder)
+    {
+        builder.Property(e => e.Email)
+            .HasMaxLength(100)
+            .IsRequired();
+
+        builder.Property(e => e.Firstname)
+            .HasMaxLength(100)
+            .IsRequired();
+
+        builder.Property(e => e.Lastname)
+            .HasMaxLength(100)
+            .IsRequired();
+
+        builder.Property(e => e.Salary)
+            .IsRequired();
+
+        builder.Property(e => e.DepartmentId)
+            .IsRequired();
+
+        builder.Property(e => e.ProjectId)
+            .IsRequired();
+        
+        builder.HasIndex(e => e.DepartmentId)
+            .IsUnique();
```

Comment:
if `Employee.DepartmentId` will be unique, then there 2 different `Employee`s could be not from the same project.

Yes, it is the index but non-unique.

Please, update this.

Reply:
fixed